test/common/request_info/dynamic_metadata_impl_test.cc

```diff
@@ -0,0 +1,131 @@
+#include "envoy/common/exception.h"
+
+#include "common/request_info/dynamic_metadata_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace RequestInfo {
+namespace {
+
+class TestStoredType {
+public:
+  TestStoredType(int value, size_t* access_count, size_t* destruction_count)
+      : value_(value), access_count_(access_count), destruction_count_(destruction_count) {}
+  ~TestStoredType() {
+    if (destruction_count_) {
+      ++*destruction_count_;
+    }
+  }
+
+  int Access() const {
```

Comment:
Does this make sense as `const` given it mutates stuff, or is access count hidden and so it's fine?

Reply:
So this strikes me as the bitwise const versus conceptually const argument, on which I've always landed on the conceptual side.  Also note that the function must be const, as dynamic_metadata() always returns a const reference for stored functions.  That + this being in a test file makes me think that this is fine.  Let me know if you disagree.